src/ossia-max/src/object_base.cpp

```diff
@@ -714,20 +664,199 @@ void object_base::push_parameter_value(ossia::net::parameter_base* param, const
+      {
+        switch(parent->m_otype)
+        {
+          case object_class::device:
+          case object_class::client:
+          {
+            auto dev = static_cast<device_base*>(parent);
+            if(dev->m_device)
+            {
+              return {std::make_shared<matcher>(&dev->m_device->get_root_node(), parent)};
+            }
+            else
+              return {};
+          }
+          case object_class::model:
+          case object_class::view:
+            return parent->m_matchers; // should we duplicate matcher and set owner here to this ???
+          default:
+            return {std::make_shared<matcher>(&ossia_max::instance().get_default_device()->get_root_node(), this)};
+        }
+      }
+      else
+      {
+        return {std::make_shared<matcher>(&ossia_max::instance().get_default_device()->get_root_node(), this)};
+      }
+    }
+  }
+  return {};
+}
+
 void object_base::update_path()
 {
-  std::string name = object_path_absolute(this);
+  std::string name;
+
+  switch(m_addr_scope)
+  {
+    case ossia::net::address_scope::absolute:
+    {
+      std::string dev_name = "*:";
+      auto parent = static_cast<device_base*>(find_parent_object());
+      if(parent)
+      {
+        if(parent->m_device)
+          dev_name = parent->m_device->get_name();
+      }
+      else
+      {
+        dev_name = ossia_max::instance().get_default_device()->get_name();
+      }
+      name = dev_name + ":" + std::string(m_name->s_name);
+      break;
+    }
+    case ossia::net::address_scope::global:
+    {
+      name = std::string(m_name->s_name);
+      break;
+    }
+    case ossia::net::address_scope::relative:
+    {
+      name = make_global_pattern();
+      break;
+    }
+  }
 
   m_is_pattern = ossia::traversal::is_pattern(name);
+  m_path = ossia::traversal::make_path(name);
+}
+
+
+std::string object_base::make_global_pattern()
+{
+  std::vector<std::string> vs;
+  vs.reserve(32);
 
-  if(m_is_pattern)
+  assert(m_name);
+  vs.push_back(m_name->s_name);
+
+  std::string device_name = ossia_max::instance().get_default_device()->get_name();
+
+  object_base* parent = find_parent_object();
+  while(parent)
   {
-    m_path = ossia::traversal::make_path(name);
+    assert(parent->m_name);
+    if(parent->m_otype == object_class::device
+    || parent->m_otype == object_class::client)
+    {
+      auto dev = static_cast<device_base*>(parent);
+      if(dev->m_device)
+      {
+        device_name = dev->m_device->get_name();
+        break;
+      }
+    }
+    vs.push_back(parent->m_name->s_name);
+    parent = parent->find_parent_object();
   }
-  else
+
+  fmt::memory_buffer absolute_path;
+  auto rit = vs.rbegin();
+  for (; rit != vs.rend(); ++rit)
   {
-    m_path = std::nullopt;
+    fmt::format_to(absolute_path, "/{}", *rit);
   }
+
+  if (vs.empty())
+    fmt::format_to(absolute_path, "/");
+
+  return device_name + ":" + std::string(absolute_path.data(), absolute_path.size());
```

Comment:
wonder if it wouldn't be more performant to do format_to( absolute_path, "{}:", device_name); instead ? IIRC it's additive

Reply:
sure, thanks for the fmt tutorial ;-)

Reply:
but a quick test shows that it returns `/blabladevice_name:` instead of `device_name:/blabla` I'll change it to add device_name before concatening the path